### PR TITLE
Upgrade S4Hana connector to IVY10, fix bugs, cleanup, write test cases

### PR DESCRIPTION
**Pull request by drag0p**
Upgrade S4Hana connector to IVY10

- fix bugs
- cleanup
- write test cases

### CODE REVIEW

File: S4HANA/processes/getHANA.p.json [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+    "data" : "com.axon.market.s4.hana.getHANAData"
+  },
+  "elements" : [ {
+      "id" : "f7",
+      "type" : "RestClientCall",
+      "config" : {
+        "path" : "/A_BusinessPartner",
+        "headers" : {
+          "Accept" : "application/json",
+          "sap-client" : "in.businessPartnerRequest.sapClient"
```
Comment (ivy-rew) [inline S4HANA/processes/getHANA.p.json:15]: from where does the use know this value? is it a static ID? or resolved from another service?
to me it looks, as if it should be designed as Re-Usable RestFeature, that add this header on demand.  ... and reads the value (200) from a the ivy.vars.

File: S4HANA/processes/getHANA.p.json [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+        "statusErrorCode" : "ivy:error:rest:client",
+        "responseMapping" : {
+          "out.businessPartners" : "result.d.results as List"
+        },
+        "templateParams" : {
+          "URI" : "in.uriPath"
+        },
+        "resultType" : "com.axon.market.s4hana.client.Wrapper9",
+        "properties" : {
+          "jersey.config.client.connectTimeout" : "hana.url.URLBusinessPartnerService.CONNECT_TIMEOUT_VALUE"
```
Comment (ivy-rew) [inline S4HANA/processes/getHANA.p.json:38]: I'd leave that out, if we need a special connection-timeout, it should be read on the global rest-client definition properties. 
I also think, that currently this only configures our default that the ivy-rest client already has '30'000' ?

File: S4HANA/src/hana/s4error/Application.java [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+import com.fasterxml.jackson.annotation.JsonAnySetter;
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.annotation.JsonPropertyOrder;
+
+@JsonInclude(JsonInclude.Include.NON_NULL)
+@JsonPropertyOrder({ "component_id", "service_namespace", "service_id", "service_version" })
+@Generated("jsonschema2pojo")
+public class Application {
```
Comment (ivy-rew) [inline S4HANA/src/hana/s4error/Application.java:16]: isn't this type outlined in an odata->openapi.json descriptor? 
If so we I'd prefer to have an openapi.client jar ... that contains that generated code.

File: S4HANA/src/hana/utils/Postman.java [hunk 4]
```diff
@@ -0,0 +1,72 @@
+package hana.utils;
+
+import java.util.Map;
+
+import javax.ws.rs.client.Entity;
+import javax.ws.rs.client.Invocation;
+import javax.ws.rs.client.WebTarget;
+import javax.ws.rs.core.Response;
+
+public class Postman {
```
Comment (ivy-rew) [inline S4HANA/src/hana/utils/Postman.java:10]: mhh what's the idea of this class?
looks as if it work around the Ivy.rest() client APIs ... and therefore benefits of none of the features in our stack: no client caching, no-dynamic variable injection, no tracing .... we should only publish such classes for very good reasons? are there any? :thinking:

File: S4HANA-product/pom.xml [hunk 5]
```diff
@@ -1,8 +1,8 @@
 <project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 https://maven.apache.org/xsd/maven-4.0.0.xsd">
   <modelVersion>4.0.0</modelVersion>
-  <groupId>com.axonivy.market</groupId>
-  <artifactId>MY-PRODUCT-NAME-product</artifactId>
-  <version>9.3.0-SNAPSHOT</version>
+  <groupId>com.axonivy.connector.s4hana</groupId>
+  <artifactId>s4-hana-product</artifactId>
+  <version>9.3.4-SNAPSHOT</version>
```
Comment (ivy-rew) [inline S4HANA-product/pom.xml:5]: we no longer offer artifacts for pre 10.0 version -> please streamline all pom versions 10.0.0-SNAPSHOT

File: pom.xml [hunk 6]
```diff
@@ -1,9 +1,9 @@
 <project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 https://maven.apache.org/xsd/maven-4.0.0.xsd">
   <modelVersion>4.0.0</modelVersion>
   <groupId>com.axonivy.market</groupId>
-  <name>my-product</name><!-- fill your product name -->
-  <artifactId>my-product-modules</artifactId><!-- fill your product name + a "-modules" postfix -->
-  <version>9.3.0-SNAPSHOT</version>
+  <name>S4HANA</name>
+  <artifactId>s4-hana-modules</artifactId>
+  <version>9.3.4-SNAPSHOT</version>
```
Comment (ivy-rew) [inline pom.xml:6]: 10

File: S4HANA/processes/getHANA.p.json (hunk 2 above)
Comment (Helmut-B) [reply S4HANA/processes/getHANA.p.json:38]: It is ok

File: S4HANA/processes/getHANA.p.json (hunk 1 above)
Comment (Helmut-B) [reply S4HANA/processes/getHANA.p.json:15]: This is standard module in SAP as far as I know.

File: pom.xml (hunk 6 above)
Comment (drag0p) [reply pom.xml:6]: version was changed to 10.0.0-SNAPSHOT

File: S4HANA-product/pom.xml (hunk 5 above)
Comment (drag0p) [reply S4HANA-product/pom.xml]: version was changed to 10.0.0-SNAPSHOT

File: S4HANA/src/hana/utils/Postman.java (hunk 4 above)
Comment (drag0p) [reply S4HANA/src/hana/utils/Postman.java:10]: I have removed the Postman class, it is probably a leftover from previous implementation

File: S4HANA/src/hana/s4error/Application.java (hunk 3 above)
Comment (drag0p) [reply S4HANA/src/hana/s4error/Application.java:16]: I have removed this s4error package

File: S4HANA/processes/getHANA.p.json (hunk 2 above)
Comment (drag0p) [reply S4HANA/processes/getHANA.p.json:38]: I have removed this connectTimeout

File: S4HANA-product/README.md [hunk 7]
```diff
@@ -0,0 +1,11 @@
+# S4HANA
+
+SAP4 HANA REST Connector
```
[the remark this answers is not in the document]
Comment (ivy-rew) [reply S4HANA-product/README.md:3]: thanks for sharing the scope of this connector :+1:

**Pull request merged**